### PR TITLE
Autosave MDZ, MFI, MMR, KFL, HMA

### PR DESCRIPTION
**Pull request by regit66**

### CODE REVIEW

**Review by abizukojc — CHANGES_REQUESTED** [review]
You have still conflicts in PR.

**Comment by regit66** [comment]
Yes, i know :)

**Comment by abizukojc** [comment]
Build on Travis failed

**Comment by abizukojc** [comment]
You have new conflicts

File: config.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#Thu Feb 16 13:11:56 CET 2017
+autosave_interval=120
+autosave_enabled=false
+autosave_path=D\:\\
```
Comment (orchowski) [inline config.properties]: I think it's wrong. My notebook have only c: partition and... what about linux users?

File: violetproduct-swing/src/test/java/com/horstmann/violet/application/autosave/AutosaveSettingsTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	public void testSetAutosaveInterval() {
+		int autoSaveInterval = 1;
+		autosaveSettings = new AutosaveSettings();
+		autosaveSettings.setAutosaveInterval(autoSaveInterval);
+		assertEquals(autoSaveInterval, autosaveSettings.getAutosaveInterval());
+	}
+
+	@Test
+	public void testGetAutosavePath() {
+		String expectedResult = "c:\\ff";
```
Comment (orchowski) [inline violetproduct-swing/src/test/java/com/horstmann/violet/application/autosave/AutosaveSettingsTest.java]: path C:\\ 
i think that on linux this tests will not pass

**Review by orchowski — CHANGES_REQUESTED** [review]

File: config.properties (hunk 1 above)
Comment (regit66) [reply config.properties]: yes , you have right

**Comment by regit66** [comment]
now should work

File: config.properties (hunk 1 above)
Comment (orchowski) [reply config.properties]: System.getProperty("user.home")   <- user home directory which is system independent

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/autosave/AutoSaveRecover_fr.properties [hunk 3]
```diff
@@ -0,0 +1,3 @@
+dialog.autosave.title=Autosave r�cup�rer
```
Comment (abizukojc) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/autosave/AutoSaveRecover_fr.properties]: For national chars use unicodes (\uXXXX)

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/autosave/AutoSaveRecover_fr.properties [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+dialog.autosave.recover=Restaurer projet non enregistr�
```
Comment (abizukojc) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/autosave/AutoSaveRecover_fr.properties]: Up

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/autosave/IAutoSave.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.horstmann.violet.application.autosave;
+
+/**
+ * AutoSave Listener
+ * 
+ * @author Marta Mrugalska
```
Comment (abizukojc) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/autosave/IAutoSave.java]: It should be removed

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/gui/SettingsDialog.java [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+import com.horstmann.violet.application.autosave.AutosaveSettings;
+import com.horstmann.violet.application.autosave.IAutoSave;
+import com.horstmann.violet.framework.injection.bean.ManiocFramework.BeanInjector;
+import com.horstmann.violet.framework.injection.resources.ResourceBundleInjector;
+import com.horstmann.violet.framework.injection.resources.annotation.ResourceBundleBean;
+
+/**
+ * Violet's settings dialog
+ * 
+ * @author Marta Mrugalska
```
Comment (abizukojc) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/gui/SettingsDialog.java]: See above

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/gui/SettingsDialog_de.properties [hunk 7]
```diff
@@ -0,0 +1,9 @@
+dialog.checkbox.enableautosaveText=Autosave aktivieren
+dialog.label.saveIntervalText=Intervall speichern
+dialog.label.secondsText=Sekunden
+dialog.label.saveDirectoryText=Verzeichnis speichern
+dialog.button.selectPathText=Pfad ausw�hlen�
```
Comment (abizukojc) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/gui/SettingsDialog_de.properties]: Encoding in this file - see above

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/gui/SettingsDialog_fr.properties [hunk 8]
```diff
@@ -0,0 +1,9 @@
+dialog.checkbox.enableautosaveText=Activer sauvegarde automatique
+dialog.label.saveIntervalText=Intervalle d'enregistrement
+dialog.label.secondsText=secondes
+dialog.label.saveDirectoryText=Enregistrer le r�pertoire
```
Comment (abizukojc) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/gui/SettingsDialog_fr.properties]: Up

**Review by abizukojc — CHANGES_REQUESTED** [review]
Check the comments

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/gui/SettingsDialog.java (hunk 6 above)
Comment (abizukojc) [reply violetproduct-swing/src/main/java/com/horstmann/violet/application/gui/SettingsDialog.java]: What with this ?

**Review by abizukojc — APPROVED** [review]

**Review by orchowski — APPROVED** [review]

**Pull request merged**